utils/Azure/acrTools.ts

```diff
@@ -0,0 +1,277 @@
+/**
+ * this function implements a new Service principal model for ACR and gets the valid login credentials to make an API call
+ * @param subscription : the subscription the registry is on
+ * @param registry : the registry to get login credentials for
+ * @param context : if command is invoked through a right click on an AzureRepositoryNode. This context has a password and username
+ */
+export async function loginCredentials(subscription: SubscriptionModels.Subscription, registry: Registry, context?: AzureImageNode | AzureRepositoryNode): Promise<{ password: string, username: string }> {
+    let node: AzureImageNode | AzureRepositoryNode;
+    if (context) {
+        node = context;
+    }
+    let username: string;
+    let password: string;
+    const client = AzureCredentialsManager.getInstance().getContainerRegistryManagementClient(subscription);
+    const resourceGroup: string = registry.id.slice(registry.id.search('resourceGroups/') + 'resourceGroups/'.length, registry.id.search('/providers/'));
+    if (context) {
+        username = node.userName;
+        password = node.password;
+    } else if (registry.adminUserEnabled) {
+        let creds = await client.registries.listCredentials(resourceGroup, registry.name);
+        password = creds.passwords[0].value;
+        username = creds.username;
+    } else {
+        //grab the access token to be used as a password, and a generic username
+        let creds = await getTokens(registry);
+        password = creds.accessToken;
+        username = '00000000-0000-0000-0000-000000000000';
+    }
+    return { password, username };
+}
+
+/**
+ *
+ * @param http_method : the http method, this function currently only uses delete
+ * @param login_server: the login server of the registry
+ * @param path : the URL path
+ * @param username : registry username, can be in generic form of 0's, used to generate authorization header
+ * @param password : registry password, can be in form of accessToken, used to generate authorization header
+ */
+export async function request_data_from_registry(http_method: string, login_server: string, path: string, username: string, password: string): Promise<void> {
+    let url: string = `https://${login_server}${path}`;
+    let header = _get_authorization_header(username, password);
+    let opt = {
+        headers: { 'Authorization': header },
+        http_method: http_method,
+        url: url
+    }
+    let err = false;
+    try {
+        let response = await request.delete(opt);
+    } catch (error) {
+        err = true;
+        console.log(error);
+    }
+    if (!err) {
+        vscode.window.showInformationMessage('Successfully deleted image');
+    }
+}
+
+/**
+ *
+ * @param registry gets the subscription for a given registry
+ * @returns a subscription object
+ */
+export function getSub(registry: Registry): SubscriptionModels.Subscription {
```

Comment:
The naming for this function remains confusing as to what it actually does. Names should be self explanatory for functions.